packages/celotool/src/e2e-tests/transfer_tests.ts

```diff
@@ -829,4 +867,78 @@ describe('Transfer tests', function(this: any) {
+        before(`start geth on sync: ${syncMode}`, () => startSyncNode(syncMode))
+
+        describe('when CeloGold is frozen', () => {
+          before('ensure gold transfers are frozen', async () => {
+            await freeze('http://localhost:8545', validatorAddress, CeloContract.GoldToken)
+          })
+
+          describe('check if frozen', () => {
+            it('should be frozen', async () => {
+              const goldTokenAddress = await kit.registry.addressFor(CeloContract.GoldToken)
+              const freezer = await kit._web3Contracts.getFreezer()
+              const isFrozen = await freezer.methods.isFrozen(goldTokenAddress).call()
+              assert(isFrozen)
+            })
+          })
+          describe('when neither sender nor receiver is whitelisted', () => {
+            before('ensure neither sender nor receiver is whitelisted', async () => {
+              await setAddressWhitelist('http://localhost:8545', validatorAddress, [])
+            })
+            it('should not add the transaction to the pool', async () => {
+              const gas = INTRINSIC_TX_GAS_COST + ADDITIONAL_INTRINSIC_TX_GAS_COST
+              const feeCurrency = await kit.registry.addressFor(CeloContract.StableToken)
+              try {
+                const res = await transferCeloGold(FromAddress, ToAddress, TransferAmount, {
+                  gas,
+                  feeCurrency,
+                })
+                await res.waitReceipt()
+                assert.fail('no error was thrown')
+              } catch (error) {
+                assert.include(error.toString(), 'Returned error: transfers are currently frozen')
+              }
+            })
+          })
+          describe('when receiver is whitelisted', () => {
+            it('should transfer succesfully', async () => {
+              const whitelistedAddress = await kit.registry.addressFor(CeloContract.LockedGold)
+              testTransferToken({
+                expectedGas: INTRINSIC_TX_GAS_COST + ADDITIONAL_INTRINSIC_TX_GAS_COST,
+                transferToken: CeloContract.GoldToken,
+                feeToken: CeloContract.GoldToken,
+                toAddress: whitelistedAddress,
+              })
+            })
+          })
+          describe('when sender is whitelisted', () => {
+            before('add sender to transfer whitelist', async () => {
+              await whitelistAddress('http://localhost:8545', validatorAddress, FromAddress)
+            })
+            it('should transfer succesfully', async () => {
+              testTransferToken({
+                expectedGas: INTRINSIC_TX_GAS_COST + ADDITIONAL_INTRINSIC_TX_GAS_COST,
+                transferToken: CeloContract.GoldToken,
+                feeToken: CeloContract.GoldToken,
+              })
+            })
+            after('remove sender from transfer whitelist', async () => {
+              await setAddressWhitelist('http://localhost:8545', validatorAddress, [])
```

Comment:
can we also test that transfers are blocked by making this change?

Reply:
We are checking that in the block described 'when neither sender nor receiver is whitelisted'

Reply:
I think it's important to test the state transition and not just the state itself. For example, if it defaults to state A (whitelist empty) and we transition to state B (whitelist populated), then we still need to test transition back to A (re-empty list). If there was a bug that allowed the whitelist to be added to but not removed from then this type of test would showcase that bug.